src/vos/tests/vts_mvcc.c

```diff
@@ -1486,6 +1492,25 @@ uncertainty_check_exec_one(struct io_test_args *arg, int i, int j, bool empty,
 		print_message("FAILED: "DF_CASE"\n",
 			      DP_CASE(i, j, empty, w, wp, we, commit, a, ap, ae,
 				      bound, mvcc_arg->i));
+
+	if (!daos_is_zero_dti(&wtx->th_saved_xid)) {
+		if (wtx->th_skip_commit)
+			vos_dtx_commit(arg->ctx.tc_co_hdl, &wtx->th_saved_xid,
+				       1, NULL);
+		else
+			vos_dtx_abort(arg->ctx.tc_co_hdl, DAOS_EPOCH_MAX,
+				      &wtx->th_saved_xid, 1);
+	}
+
+	if (!daos_is_zero_dti(&atx->th_saved_xid)) {
+		if (atx->th_skip_commit)
+			vos_dtx_commit(arg->ctx.tc_co_hdl, &atx->th_saved_xid,
+				       1, NULL);
+		else
+			vos_dtx_abort(arg->ctx.tc_co_hdl, DAOS_EPOCH_MAX,
```

Comment:
did the semantics change?  I did abort before but it caused issues if the tx failed.

Reply:
In my test, I found that some old DTX entries affect the subsequent tests, but I cannot exactly point out which test cases left them, then I make a generally cleanup before each test case exit to guarantee that next run will be in a clean environment.